src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -1403,7 +1471,7 @@ public static void ChangePassword(string connectionString, SqlCredential credent
                 throw ADP.InvalidArgumentLength(nameof(newSecurePassword), TdsEnums.MAXLEN_NEWPASSWORD);
             }
 
-            SqlConnectionPoolKey key = new SqlConnectionPoolKey(connectionString, credential);
+            SqlConnectionPoolKey key = new SqlConnectionPoolKey(connectionString, credential: null, accessToken: null);
```

Comment:
Was changing credential to null here expected? There is another similar spot below on line 1512 as well.

Reply:
/cc: @danmosemsft / @stephentoub for visibility

Reply:
@afsanehR @keeratsingh own this

Reply:
Line 1512 has a bug and credential shouldn't be passed in as null, instead of the credential parameter should be passed through.

Reply:
Also this line, i.e. 1474, its buggy. No null should be passed in for Credential